File: BackendAPI/Application/DTOs/Output/AccountOutputDto.cs
using Domain.Enums;

namespace Application.DTOs.Output;

public class AccountOutputDto
{
    public required AccountType AccountType { get; set; }
    public required string Email { get; set; }

    // Common fields
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Telephone { get; set; }

    // Kweker specific
    public string? CompanyName { get; set; }
    public string? KvkNumber { get; set; }

    // Veilingmeester specific
    public string? CountryCode { get; set; }
    public string? Region { get; set; }
    public string? AuthorisatieCode { get; set; }

    // Address (Primary for Koper, Main for Kweker)
    public AddressOutputDto? Address { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/AccountOutputDto.ts
import { AccountType } from '../../enums/AccountTypes';
import { AddressOutputDto } from './AddressOutputDto';

// AccountOutputDto.ts
export interface AccountOutputDto {
	accountType: AccountType;
	email: string;
	firstName?: string;
	lastName?: string;
	telephone?: string;
	companyName?: string;
	kvkNumber?: string;
	countryCode?: string;
	region?: string;
	address?: AddressOutputDto;
	primaryAddressId?: number;
	addresses?: AddressOutputDto[];
}
